lesson-7/ReportService.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.IO;
+using TemplateEngine.Docx;
+
+namespace lesson_7
+{
+    public sealed class ReportService
+    {
+        public void GenerateReport(CompanyInfo companyInfo, string output = "")
+        {
+            if (companyInfo is null)
+            {
+                return;
+            }
+
+            if (string.IsNullOrWhiteSpace(output))
+            {
+                output = Path.Combine(Directory.GetCurrentDirectory(), "CompanyReport.docx");
+            }
```

Comment:
```suggestion
        public void GenerateReport(CompanyInfo companyInfo, string output = null)
        {
            if (string.IsNullOrWhiteSpace(output))
            {
                output = Path.Combine(Directory.GetCurrentDirectory(), "CompanyReport.docx");
            }
```

двойная проверка одного параметра на null бесполезна